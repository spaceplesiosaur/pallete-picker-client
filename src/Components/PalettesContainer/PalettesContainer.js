import React from 'react';
import './PalettesContainer.scss';

import Palette from '../Palette/Palette';
import SavePaletteForm from '../SavePaletteForm/SavePaletteForm';
import GeneratePalette from '../GeneratePalette/GeneratePalette';

const PalettesContainer = () => {

  const colorListState = useState([{name: "color1", color: "", frozen: false}, {name: "color2", color: "", frozen: false}, {name: "color3", color: "", frozen: false}, {name: "color4", color: "", frozen: false}, {name: "color5", color: "", frozen: false} ])
  const colorList = colorListState[0]
  const changeColor = colorListState[1]

  const freezeColor = (name) => {
    const updatedColors = colorList.forEach(color => {
      if (color.name === name) {
        color.frozen = !color.frozen
      }
    })
    changeColor(updatedColors)
  }

  const makeColors = () => {
    return colorList.map((palette) => {
      <Palette
      key={palette.name}
      freezeColor={freezeColor}
      {...palette}
      />
    })
  }

    return (
      <section className='palette-container'>
          <div className='generate-pal'>
            {makeColors()}
            <GeneratePalette />
          </div>
         <SavePaletteForm />
      </section>
    );
}

export default PalettesContainer;
