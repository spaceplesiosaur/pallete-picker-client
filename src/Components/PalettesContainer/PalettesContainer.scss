.palette-container {
    width: 50%;
    background-color: #80008091;
    margin: 20px;
    border-radius: 20px;
    border-right: 3px solid #79189D;
    .generate-pal {
        display: flex;
        justify-content: space-around;
        margin: 20px;
        .pal {
            width: 47%;
        }
    }
}