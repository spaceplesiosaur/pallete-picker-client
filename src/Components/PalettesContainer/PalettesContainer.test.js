import React from 'react';
import { shallow } from 'enzyme';
import PalettesContainer from './PalettesContainer'

describe('PalettesContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PalettesContainer />)

  })

  it('should render with the correct properites', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
