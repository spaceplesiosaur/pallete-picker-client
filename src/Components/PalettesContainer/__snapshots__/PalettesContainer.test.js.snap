// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PalettesContainer should render with the correct properites 1`] = `
<section
  className="palette-container"
>
  <div
    className="generate-pal"
  >
    <div
      className="pal"
    >
      <Palette
        color="#B42D91"
        freezeColor={[Function]}
        frozen={false}
        key="color1"
        name="color1"
      />
      <Palette
        color="#890CA6"
        freezeColor={[Function]}
        frozen={false}
        key="color2"
        name="color2"
      />
      <Palette
        color="#B42D91"
        freezeColor={[Function]}
        frozen={false}
        key="color3"
        name="color3"
      />
      <Palette
        color="#890CA6"
        freezeColor={[Function]}
        frozen={false}
        key="color4"
        name="color4"
      />
      <Palette
        color="#B42D91"
        freezeColor={[Function]}
        frozen={false}
        key="color5"
        name="color5"
      />
    </div>
    <GeneratePalette
      randomizeColors={[Function]}
    />
  </div>
  <Connect(SavePaletteForm)
    colorList={
      Array [
        Object {
          "color": "#B42D91",
          "frozen": false,
          "name": "color1",
        },
        Object {
          "color": "#890CA6",
          "frozen": false,
          "name": "color2",
        },
        Object {
          "color": "#B42D91",
          "frozen": false,
          "name": "color3",
        },
        Object {
          "color": "#890CA6",
          "frozen": false,
          "name": "color4",
        },
        Object {
          "color": "#B42D91",
          "frozen": false,
          "name": "color5",
        },
      ]
    }
  />
</section>
`;
